### ISSUE TITLE
Training speed too slow

### ISSUE THREAD

**Issue by deli084618** [NVIDIA/tacotron2#59]
My training speed is about 5s / iteration. Is that normal (Single GPU)? I tried training with 4 GPU but the speed seems not change much.

I noticed that there is load_mel_from_disk option which may be the key factor of training speed. Can I use the preprocess.py from https://github.com/r9y9/wavenet_vocoder/ to generate mels to be loaded?

Besides, I multiplied the learning rate by 0.01 after 150000 steps, however, the validation loss seems not decreasing. The audio generated by GL_vocoder isn't clear enough like demo.wav.
[tacotron_audio.wav.zip](https://github.com/NVIDIA/tacotron2/files/2187753/tacotron_audio.wav.zip)
Actually, the quality of audio is not improving after 150000 steps.

The images are shown below. Any guidance or suggestions would be appreciated.
![training_loss](https://user-images.githubusercontent.com/32466837/42621808-62bc9a62-85f1-11e8-87a8-187062782866.PNG)
![validation_loss](https://user-images.githubusercontent.com/32466837/42621810-631b430a-85f1-11e8-9588-da0b05525b8e.PNG)
![grad norm](https://user-images.githubusercontent.com/32466837/42621833-724bb6ca-85f1-11e8-8638-d5836e373258.PNG)
![alignment](https://user-images.githubusercontent.com/32466837/42621837-74db28da-85f1-11e8-9489-59eeb55a1517.png)
Mel Predicted
![mel_predicted](https://user-images.githubusercontent.com/32466837/42621842-776046ee-85f1-11e8-8ceb-af57026f60d9.png)
Target Mel
![mel_target](https://user-images.githubusercontent.com/32466837/42621843-79284166-85f1-11e8-9575-f8ee5cb9a6fa.png)

**Comment by gsoul** [comment]
In terms of the number of GPUs and speed I think that's ok, as what happens there is that you put batch_size samples to each GPU simultaneously, so instead of calculating gradient from batch_size samples it is calculated from n_gpus*batch_size which might lead to smaller variation in gradients.

**Comment by rafaelvalle** [comment]
@deli084618 the plots you shared look good overall. 
The fact that the speed is the same with 4 GPUs is great because it means you were able to increase your batch size by 4 without any time performance penalty. 

If you choose to use pre-processing from Ryuchi's repo, make sure you have the same mel-spectrogram specifications in every step.

Audio quality is extremely dependent on the Mel2Audio decoder. You can improve the audio quality of your sample by using a Wavenet decoder instead of GL.

Check this https://github.com/NVIDIA/tacotron2/issues/52 for guidance on how to use a pre-trained Wavenet.

**Comment by rafaelvalle** [comment]
Closing due to inactivity.
